Automattic/wp-calypso
# Jetpack plans: remove component that is not in use anymore

The old jetpack-plan-details component have been replaced by https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/upgrades/checkout-thank-you/jetpack-plan-details.jsx, but the code is still around. This PR removes it.

how to test
==========
1. You need a connected jetpack site. Purchase a plan for it and make sure that you still see the thank you page

ping  @rickybanister  because he worked on that original component, and @dzver , who discovered this dead code :)
